test/support/factories.ex

```diff
@@ -251,7 +251,8 @@ defmodule Lightning.Factories do
   end
 
   def simple_workflow_factory(attrs) do
-    trigger = build(:trigger, type: :webhook)
+    {trigger_enabled, attrs} = Map.pop(attrs, :trigger_enabled, true)
+    trigger = build(:trigger, type: :webhook, enabled: trigger_enabled)
```

Comment:
I'm not sure I'm crazy about extending the `attrs` interface here.
I'd like to avoid adding keys that only exist on the factory.

I propose changing the tests where we need to control the triggers attrs using the existing functions:

```
build(:workflow)
|> with_trigger(trigger)
|> with_job(job)
|> with_edge({trigger, job})
|> insert()
```

But I do agree that we should revisit this pattern when we find ourselves repeating this more often.

Reply:
Wanted to continue using the `simple_worfklow` which wasn't accepting a `triggers` but a trigger on the test setup was enough. Thanks!